machine.go

```diff
@@ -126,6 +126,17 @@ func (cfg *Config) Validate() error {
 		return fmt.Errorf("socket %s already exists", cfg.SocketPath)
 	}
 
+	if cfg.MachineCfg.VcpuCount == nil ||
```

Comment:
I'd replace these conditionals with `assert.GreaterOrEqual(t, cfg.MachineCfg.VcpuCount, 1, "machine needs a nonzero VcpuCount")` and `assert.NotNil(t, cfg.MachineCfg.HtEnabled, "machine needs a setting for ht_enabled")`.

Reply:
This is in the actually validation code, not in the test code. So the right behavior is to return a validation failure error to the calling code.

Reply:
Thanks, you are correct.